js/app/modules/selection/xapian.js

```diff
@@ -84,30 +84,50 @@ const Xapian = new Module.Class({
                 }
                 return;
             }
-            let more = info.more_results;
 
-            if (!more) {
+            let results_added = results.filter(this.add_model, this);
+            let need_more_results = num_desired > results_added.length;
```

Comment:
Yea so basically what I'll be changing is this line, so that we don't call `add_model` on _more_ results than was originally requested by `num_desired`